spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
+        deps = requires + optional
+
+        if plugin_name in self._PLUGINS:
+            for name, plugin_instance in self._PLUGINS.items():
+                if name == plugin_name and name in deps:
+                    return plugin_instance
+            else:
+                raise Exception(
+                    'Plugin "{}" not found!'.format(plugin_name))
+        else:
+            raise Exception('Plugin "{}" not part of REQUIRES or OPTIONAL '
+                            'requirements!'.format(plugin_name))
+
+    def options_from_conf(self, options):
+        """
+        Get the options from the configurations files.
+
+        Returns
+        -------
+        Dictionary of {str: object}
+
+        """
+        config_options = {}
+        for option in options:
+            try:
+                value = self.get_option(option)
+            except Exception:
+                # TODO: Remove when migration is done, move to logger
+                print('Warning: option "{}" not found in section "{}" '
+                      'of configuration!'.format(option, self.NAME))
+                value = options[option]
+
+            config_options[option] = value
+
+        return config_options
+
+    def get_option(self, option, default=NoDefault, section=None):
+        """
+        Get an option from Spyder configuration files.
+
+        Parameters
+        ----------
+        option: str
+            Name of the option to get its value from.
+        default: bool, int, str, tuple, list, dict, NoDefault
+            Value to save in configuration file, passed as a Python
+            object.
+        section: str
+            TODO:
+
+        Returns
+        -------
+        bool, int, str, tuple, list, dict
+            Value associated with `option`.
+        """
+        if self._conf is not None:
+            section = self.CONF_SECTION if section is None else section
+            return self._conf.get(section, option, default)
+
+    @Slot(str, object)
+    @Slot(str, object, str)
+    def set_option(self, option, value, section=None):
+        """
+        Set an option in Spyder configuration files.
+
+        Parameters
+        ----------
+        option: str
+            Name of the option (e.g. 'case_sensitive')
+        value: bool, int, str, tuple, list, dict
+            Value to save in configuration file, passed as a Python
+            object.
+        section: str
+            TODO
+
+        Notes
+        -----
+        * Use sig_option_changed to call this method from widgets of the
+          same or another plugin.
+        * CONF_SECTION needs to be defined for this to work.
+        """
+        if self._conf is not None:
+            section = self.CONF_SECTION if section is None else section
+            self._conf.set(section, str(option), value)
+
+    # TODO: Configuration vs Options vs settings?
+    # We could use options for widgets and decide on configuration vs
+    # settings
```

Comment:
You could change this method's name to `apply_conf` or `apply_config`, to be more consistent.

Reply:
Fixed